src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach(item => delete cloneState[item]);
+        break;
+
+      case 'clear':
+        for (const key in cloneState) {
+          delete cloneState[key];
+        }
+        break;
+
+      default:
+        break;
+    }
+
+    result.push({ ...cloneState });
```

Comment:
Instead of pushing a clone of 'cloneState' to 'result' after every action, you can push the current state at the beginning of each iteration. This will ensure that the initial state is also included in your 'result' array.

Reply:
fixed